exercises/concept/calculator-conundrum/.meta/src/reference/java/CalculatorConundrum.java

```diff
@@ -0,0 +1,30 @@
+public class CalculatorConundrum {
+    private final String answer;
+
+    public CalculatorConundrum(int operand1, int operand2, String operation) {
+        int result;
+        try {
+            if (operation.equals("+")) {
+                result = SimpleOperation.addition(operand1, operand2);
+            } else if (operation.equals("*")) {
+                result = SimpleOperation.multiplication(operand1, operand2);
+            } else if (operation.equals("/")) {
+                result = SimpleOperation.division(operand1, operand2);
+            } else if (operation.equals("")) {
+                throw new IllegalArgumentException("Operation cannot be empty");
+            } else if (operation == null) {
+                throw new NullPointerException("Operation cannot be null.");
+            } else {
+                throw new IllegalArgumentException(String.format("%s operation does not exist", operation));
+            }
+        } catch (ArithmeticException e){
+            throw new ArithmeticException(e.getMessage());
+        }
+        this.answer = String.format("%d %s %d = %s", operand1, operation, operand2, result);
+    }
+
+    public String calculate() {
+        return answer;
```

Comment:
To be honest, it makes more sense to me for the constructor to just hold the entries and then for the `calculate` method to actually apply the operation (throwing exceptions when necessary).

Reply:
I recently had some exercise reviewed where they said to do the process part inside constructor so that `calculate()` does not have to redo the same process again and again.

Reply:
There is certainly a trade-off and honestly, I would normally expect the calculate method to be the one that takes all of the input (both the operands and the operation). Maybe we should just do that:

```
public class CalculatorConundrum {
    public String calculate(int operand1, int operand2, String operation) { ... }
}
```